### ISSUE TITLE
381 instances of "C2719" (MSVC-only errors) with CMake 

### ISSUE THREAD

**Issue by solvingj** [abseil/abseil-cpp#93]
Here is an example of the errors: 

`C2719: 'other': formal parameter with requested alignment of 16 won't be aligned [C:\Users\appveyor\.conan\data\abseil\20180208\bincrafters\stable\build\349406f0066fed73a99d8a5bff4fcdfd4b07107f\source_subfolder\absl\numeric\absl_int128.vcxproj]`

CI Log:
https://ci.appveyor.com/project/BinCrafters/conan-abseil/build/1.0.68/job/auprrao9nr9phetm#L1682

It seems related to treating warnings as errors, and so we suppressed that behavior which fixed MSVC15 but not MSVC14. 

```
if(MSVC)
  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /wd4309")
  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /wd4309")
endif()
```

I found an issue where I believe the vcpkg team hit the same problem and then just patched the sources to get around it: 
https://github.com/Microsoft/vcpkg/issues/1978

If anyone has any ideas or experience with this, would be great to hear.

**Comment by astrelni** [comment]
Is this a 32-bit build? My working hypothesis is that this is a 32-bit build which means alignof(max_align_t) is 8, and the request for 16 byte alignment is not supported. I will get to work on a fix.

**Comment by solvingj** [comment]
Indeed it is, thanks so much!
```
[settings]
arch=x86
build_type=Release
compiler=Visual Studio
compiler.runtime=MT
compiler.version=14
```

**Comment by solvingj** [comment]
@astrelni can you confirm that is likely the cause?

**Comment by astrelni** [comment]
@solvingj Yes, to the best of my knowledge that is the issue. A fix went in yesterday that will come out with the daily release either today or Monday that removes the alignment specifier on platforms that do not have __int128. This incidentally should also solve the issue here since MSVC does not have an __int128.

**Comment by solvingj** [comment]
Wow, that's really awesome and fast turnaround.  This project is great :) Thanks for the feedback.

**Comment by derekmauro** [comment]
The fix is in 0d40cb771eec8741f44e5979cfccf1eeeedb012a. Let us know if you are still having problems.

**Comment by solvingj** [comment]
looks good, all green
